src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java

```diff
@@ -157,17 +157,19 @@ private static void destroy(String container, Launcher launcher, Node node, EnvV
 
         private static final long serialVersionUID = 1;
         private final String container;
+        private final String userId;
         private final String[] envHost;
 
-        Decorator(String container, EnvVars envHost) {
+        Decorator(String container, String userId, EnvVars envHost) {
             this.container = container;
+            this.userId = userId;
             this.envHost = Util.mapToEnv(envHost);
         }
 
         @Override public Launcher decorate(Launcher launcher, Node node) {
             return new Launcher.DecoratedLauncher(launcher) {
                 @Override public Proc launch(Launcher.ProcStarter starter) throws IOException {
-                    List<String> prefix = new ArrayList<String>(Arrays.asList("docker", "exec", container, "env"));
+                    List<String> prefix = new ArrayList<String>(Arrays.asList("docker", "exec", "--user", userId, container, "env"));
```

Comment:
Settings compatibility issue for a build started with the old plugin and resumed with the new: will pass `--user null` (I suppose). Safer to check for null here.